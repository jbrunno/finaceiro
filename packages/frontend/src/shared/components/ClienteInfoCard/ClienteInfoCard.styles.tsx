import { styled, css } from '@mui/material/styles';
import {
  Card,
  CardActions,
  CardContent,
  Chip,
  Typography,
} from '@frontend/components';

export const ChipStyled = styled(Chip)`
  margin-right: ${({ theme }) => theme.fn.spacing(5)};
`;

export const ChipNegativado = styled(Chip)(
  ({ theme }) => css`
    background-color: ${theme.palette.error.main};
    color: ${theme.palette.info.contrastText};
  `,
);

export const TelefoneTypography = styled(Typography)`
  color: ${({ theme }) => theme.palette.primary.main};
`;

export const CardActionsStyled = styled(CardActions)`
  justify-content: center;
  border-top: 1px solid ${({ theme }) => theme.palette.divider};
`;

export const CardStyled = styled(Card)`
  display: grid;
  grid-template-rows: auto 1fr auto;
`;

export const CardContentStyled = styled(CardContent)(
  ({ theme }) => css`
    display: grid;
    grid-gap: ${theme.fn.spacing(4)};
    padding-top: 0;
  `,
);
