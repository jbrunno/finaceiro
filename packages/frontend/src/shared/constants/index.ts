export const DEFAULT_SNACKBAR_TIMER = 4000;

export enum ClienteSexoEnum {
  MASCULINO = 'MASCULINO',
  FEMININO = 'FEMININO',
}

export const CLIENTE_SEXO = {
  [ClienteSexoEnum.MASCULINO]: 'Masculino',
  [ClienteSexoEnum.FEMININO]: 'Feminino',
};

export const OptionsSexo = Object.keys(CLIENTE_SEXO).map((key) => ({
  label: CLIENTE_SEXO[key as keyof typeof ClienteSexoEnum],
  value: key,
}));

export enum ContatoClassificacaoEnum {
  INDEFINIDO = 'INDEFINIDO',
  RUIM = 'RUIM',
  BOM = 'BOM',
  EXCELENTE = 'EXCELENTE',
}

export const CONTATO_CLASSIFICACAO = {
  [ContatoClassificacaoEnum.INDEFINIDO]: 'Indefinido',
  [ContatoClassificacaoEnum.RUIM]: 'Ruim',
  [ContatoClassificacaoEnum.BOM]: 'Bom',
  [ContatoClassificacaoEnum.EXCELENTE]: 'Excelente',
};

export const CONTATO_CLASSIFICACAO_STARS = [
  ContatoClassificacaoEnum.INDEFINIDO,
  ContatoClassificacaoEnum.RUIM,
  ContatoClassificacaoEnum.BOM,
  ContatoClassificacaoEnum.EXCELENTE,
];

export enum ContatoMarcadorEnum {
  PESSOAL = 'PESSOAL',
  TRABALHO = 'TRABALHO',
  AVOS = 'AVOS',
  FILHO = 'FILHO',
  MAE = 'MAE',
  PAI = 'PAI',
  NETO = 'NETO',
  SOBRINHO = 'SOBRINHO',
  TIO = 'TIO',
}

export const CONTATO_MARCADOR = {
  [ContatoMarcadorEnum.PESSOAL]: 'Pessoal',
  [ContatoMarcadorEnum.TRABALHO]: 'Trabalho',
  [ContatoMarcadorEnum.AVOS]: 'Avô / Avó',
  [ContatoMarcadorEnum.FILHO]: 'Filho(a)',
  [ContatoMarcadorEnum.MAE]: 'Mãe',
  [ContatoMarcadorEnum.PAI]: 'Pai',
  [ContatoMarcadorEnum.NETO]: 'Neto(a)',
  [ContatoMarcadorEnum.SOBRINHO]: 'Sobrinho(a)',
  [ContatoMarcadorEnum.TIO]: 'Tio(a)',
};

export enum ClienteTipoEnum {
  FISICA = 'FISICA',
  JURIDICA = 'JURIDICA',
  ESTRANGEIRA = 'ESTRANGEIRA',
  INDEFINIDO = 'INDEFINIDO',
}

export enum ERRORS_API {
  ERROR_401 = 401,
  ERROR_403 = 403,
  ERROR_406 = 'Esse dado já está cadastrado.',
}
