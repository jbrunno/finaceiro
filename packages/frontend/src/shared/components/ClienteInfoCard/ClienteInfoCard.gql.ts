import { ClienteSexoEnum, ClienteTipoEnum } from '@/shared/constants';
import { gql } from '@apollo/client';

export const CLIENTE_INFO_FRAGMENT = {
  cliente: gql`
    fragment CLIENTE_INFO_FRAGMENT on Cliente {
      id
      nome
      nomeSocial
      documento
      sexo
      dataNascimento
      cbo {
        codigo
        id
        ocupacao
      }
      falecido
      negativado
      tipo
    }
  `,
};

export type ClienteInfoData = {
  id: string;
  nome: string;
  nomeSocial: string;
  documento: string;
  sexo: ClienteSexoEnum;
  dataNascimento: string;
  cbo: {
    id: string;
    codigo: string;
    ocupacao: string;
  } | null;
  falecido: boolean;
  negativado: boolean;
  tipo: ClienteTipoEnum;
};
