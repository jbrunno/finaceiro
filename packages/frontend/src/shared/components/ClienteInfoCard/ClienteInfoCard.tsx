import { Edit as EditIcon } from '@frontend/icons';
import { useState } from 'react';
import {
  Button,
  CardHeaderChip,
  CardSkeleton,
  CardText,
  Dialog,
  Grid,
  RenderComponents,
  Typography,
} from '@frontend/components';
import { CLIENTE_SEXO, ContatoClassificacaoEnum } from '@/shared/constants';
import {
  DateOnly,
  formatDate,
  getAgeFromBirthdate,
  formatDocument,
} from '@frontend/utils';
import { toast } from 'react-toastify';
import { useWalletPermissions } from '@frontend/hooks';
import { parsePhoneNumber } from 'awesome-phonenumber';
import { ContatoRating } from '../ContatoRating/ContatoRating';
import {
  CardActionsStyled,
  CardContentStyled,
  CardStyled,
  ChipNegativado,
  ChipStyled,
  TelefoneTypography,
} from './ClienteInfoCard.styles';
import { ClienteEditInfo } from './components/ClienteEditInfo/ClienteEditInfo';
import { ClienteInfoData, CLIENTE_INFO_FRAGMENT } from './ClienteInfoCard.gql';

type ContatoClienteInfoData = {
  numero: string;
  classificacao: ContatoClassificacaoEnum;
};

type ClienteInfoCardProps = {
  screen: string;
  cliente?: ClienteInfoData | null;
  contato?: ContatoClienteInfoData;
  className?: string;
};

export function ClienteInfoCard({
  cliente,
  className,
  contato,
  screen,
}: ClienteInfoCardProps) {
  if (!cliente) {
    return <CardSkeleton />;
  }

  const [isModalOpen, setModalAsOpen] = useState<boolean>(false);
  const clienteDataNascimento = new DateOnly(cliente.dataNascimento);
  const clienteIdade = getAgeFromBirthdate(clienteDataNascimento);
  const { walletPermissions } = useWalletPermissions(screen);

  return (
    <>
      <CardStyled className={className}>
        <CardHeaderChip
          title={<Typography variant="h5">Informações do cliente</Typography>}
          chip={
            <>
              {cliente.falecido ? <ChipStyled label="Falecido" /> : null}
              {cliente.negativado ? (
                <ChipNegativado label="Negativado" />
              ) : null}
            </>
          }
        />
        <CardContentStyled>
          {cliente.nomeSocial &&
            walletPermissions.includes('CLIENTE_INFO_NOME_SOCIAL') && (
              <CardText label="Nome social" text={cliente.nomeSocial} />
            )}
          <RenderComponents screen={screen}>
            <CardText
              label="Nome de registro"
              w-perm="CLIENTE_INFO_NOME_REGISTRO"
              text={cliente.nome}
            />
            <CardText
              label="CPF/CNPJ"
              w-perm="CLIENTE_INFO_DOCUMENTO"
              text={cliente.documento ? formatDocument(cliente.documento) : '-'}
            />
            <CardText
              label="CPF/CNPJ/ID ou Passaporte"
              w-perm="CLIENTE_INFO_DOCUMENTO_INTERNACIONAL"
              text={
                cliente.documento
                  ? formatDocument(cliente.documento, cliente.tipo)
                  : '-'
              }
            />
          </RenderComponents>
          {clienteDataNascimento &&
            walletPermissions.includes(
              'CLIENTE_INFO_DATA_NASCIMENTO_IDADE',
            ) && (
              <CardText
                label="Nascimento - Idade"
                text={`${formatDate(
                  clienteDataNascimento,
                )} - ${clienteIdade} anos`}
              />
            )}
          {walletPermissions.includes('CLIENTE_INFO_SEXO') && (
            <CardText label="Sexo" text={CLIENTE_SEXO[cliente.sexo]} />
          )}

          {cliente?.cbo &&
            walletPermissions.includes('CLIENTE_INFO_PROFISSAO') && (
              <CardText label="Profissão" text={cliente.cbo.ocupacao} />
            )}
          {contato && walletPermissions.includes('CLIENTE_INFO_CONTATO') && (
            <Grid container spacing={1}>
              <Grid item xs={6}>
                <CardText
                  label="Telefone de contato atual"
                  text={
                    <TelefoneTypography>
                      {parsePhoneNumber(contato.numero).number?.international}
                    </TelefoneTypography>
                  }
                />
              </Grid>
              <Grid item xs={6}>
                <CardText
                  label="Classificação"
                  text={<ContatoRating value={contato.classificacao} />}
                />
              </Grid>
            </Grid>
          )}
        </CardContentStyled>
        <CardActionsStyled>
          {cliente && (
            <Button
              variant="outlined"
              color="primary"
              startIcon={<EditIcon />}
              onClick={() => setModalAsOpen(true)}
            >
              Editar informações
            </Button>
          )}
        </CardActionsStyled>
      </CardStyled>
      <Dialog open={isModalOpen} onClose={() => setModalAsOpen(false)}>
        <ClienteEditInfo
          screen={screen}
          cliente={cliente}
          onSuccess={() => {
            toast.success('Informações básicas alteradas com sucesso!');
            setModalAsOpen(false);
          }}
        />
      </Dialog>
    </>
  );
}

ClienteInfoCard.fragments = CLIENTE_INFO_FRAGMENT;
