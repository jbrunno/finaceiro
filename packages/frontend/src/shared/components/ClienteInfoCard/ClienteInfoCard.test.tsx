import { render } from '@test-utils/component-testing';
import { ClienteSexoEnum, ClienteTipoEnum } from '@/shared/constants';
import { ClienteInfoCard } from './ClienteInfoCard';

const dataMock = new Date('2021-01-01T00:00:00.000Z').toISOString();

const mocks = {
  id: 'teste',
  nome: 'teste',
  nomeSocial: 'teste',
  documento: '11122233344',
  sexo: ClienteSexoEnum.FEMININO,
  dataNascimento: dataMock,
  cbo: {
    id: 'teste',
    codigo: 'teste',
    ocupacao: 'teste',
  },
  falecido: false,
  negativado: false,
  tipo: ClienteTipoEnum.ESTRANGEIRA,
};

test('should render ClienteInfoCard', () => {
  const { container } = render(
    <ClienteInfoCard screen="test" cliente={mocks} />,
  );
  expect(container).toMatchSnapshot();
});
