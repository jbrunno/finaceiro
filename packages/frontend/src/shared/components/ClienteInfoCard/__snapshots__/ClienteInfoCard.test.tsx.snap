// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ClienteInfoCard 1`] = `
.emotion-2 {
  background-color: #fff;
  color: #3B3B41DE;
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  overflow: visible;
  border-radius: 0.5rem;
  box-shadow: 0px 2px 4px -1px rgba(233, 233, 233, 0.8),0px 4px 5px rgba(233, 233, 233, 0.14),0px 1px 10px rgba(233, 233, 233, 0.12);
  background-color: #fff;
  display: grid;
  grid-template-rows: auto 1fr auto;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
  display: grid;
  grid-template-columns: 1fr auto auto;
  grid-template-areas: 'content avatar action';
  border-bottom: 1px solid #0000001F;
}

.emotion-4 .MuiCardHeader-content {
  grid-area: content;
}

.emotion-4 .MuiCardHeader-avatar {
  margin-right: 0;
  margin-left: 0.5rem;
  grid-area: avatar;
}

.emotion-4 .MuiCardHeader-avatar>div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 .MuiCardHeader-action {
  grid-area: action;
  margin: auto;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-right: 16px;
}

.emotion-6 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-7 {
  margin: 0;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.125rem;
  letter-spacing: 0.0094rem;
  font-family: Lato,sans-serif;
  display: block;
}

.emotion-8 {
  display: grid;
  -webkit-column-gap: 0.25rem;
  column-gap: 0.25rem;
  white-space: nowrap;
  grid-template-columns: auto 1fr;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-column-gap: 0.25rem;
  column-gap: 0.25rem;
}

.emotion-12 {
  margin: 0;
  font-weight: 700;
  font-size: 1rem;
  line-height: 1.5rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
}

.emotion-14 {
  padding: 16px;
  display: grid;
  grid-gap: 0.25rem;
  padding-top: 0;
}

.emotion-14:last-child {
  padding-bottom: 24px;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-top: 1px solid #0000001F;
}

.emotion-16>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: Lato,sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 5px 15px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border: 1px solid rgba(74, 49, 255, 0.5);
  color: #4A31FF;
  text-transform: none;
}

.emotion-18::-moz-focus-inner {
  border-style: none;
}

.emotion-18.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-18 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-18:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(74, 49, 255, 0.04);
  border: 1px solid #4A31FF;
}

@media (hover: none) {
  .emotion-18:hover {
    background-color: transparent;
  }
}

.emotion-18.Mui-disabled {
  color: #00000042;
  border: 1px solid #0000001F;
}

.emotion-19 {
  display: inherit;
  margin-right: 8px;
  margin-left: -4px;
}

.emotion-19>*:nth-of-type(1) {
  font-size: 20px;
}

.emotion-20 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-21 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0 emotion-1 emotion-2"
  >
    <div
      class="MuiCardHeader-root emotion-3 emotion-4"
    >
      <div
        class="MuiCardHeader-avatar emotion-5"
      />
      <div
        class="MuiCardHeader-content emotion-6"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title emotion-7"
        >
          <div
            class="emotion-8 emotion-9"
          >
            <div
              class="emotion-10 emotion-11"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 emotion-12"
              >
                Informações do cliente
              </h5>
            </div>
          </div>
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root emotion-13 emotion-14"
    />
    <div
      class="MuiCardActions-root MuiCardActions-spacing emotion-15 emotion-16"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium emotion-17 emotion-18"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium emotion-19"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-20"
            data-testid="EditIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        Editar informações
        <span
          class="MuiTouchRipple-root emotion-21"
        />
      </button>
    </div>
  </div>
</div>
`;
